src/main/java/io/wisetime/idea/branch/BranchHelper.java

```diff
@@ -24,12 +26,19 @@ public String getCurrentBranchName() {
 
   public void onBranchChanged(String branchName) {
     currentBranchName = branchName;
-    updateFrameTitle();
+    ApplicationManager.getApplication().invokeLater(this::updateFrameTitle);
```

Comment:
Frame/project name changes should be done on UI thread. Sometimes git plugin sends branch change notifications from worker thread.

Reply:
This should fix #5

Reply:
this non-blocking seems cleaner